src/server/page_server.hpp

```diff
@@ -110,18 +111,51 @@ class page_server
 	// function that will, at some moment, remove that hook
 	mutable std::mutex m_prior_sender_sink_removal_mtx;
 
+	std::mutex m_abnormal_tran_server_disconnect_mtx;
 	bool m_abnormal_tran_server_disconnect;
     };
 
+    /* helper class with the task of destroying connnection handlers and, by this,
+     * also waiting for the receive and transmit threads inside the handlers to terminate
+     */
+    class disconnect_handler
+    {
+      public:
+	disconnect_handler ();
+
+	disconnect_handler (const disconnect_handler &) = delete;
+	disconnect_handler (disconnect_handler &&) = delete;
+
+	~disconnect_handler ();
+
+	disconnect_handler &operator = (const disconnect_handler &) = delete;
+	disconnect_handler &operator = (disconnect_handler &&) = delete;
+
+	void disconnect (connection_handler_uptr_t &&handler);
+	void terminate ();
+
+      private:
+	void disconnect_loop ();
+
+      private:
+	bool m_terminate = false;
```

Comment:
Shouldn't be an std::atomic<bool> m_terminate here?

Reply:
True.